### ISSUE TITLE
tsc --init update 2024

### ISSUE THREAD

**Issue by RyanCavanaugh** [microsoft/TypeScript#58420]
### Acknowledgement

- [X] I acknowledge that issues using this template may be closed without further explanation at the maintainer's discretion.

### Comment

Following up from #58417

Note: This is only for *argumentless* `tsc --init`. We know that it's physically possible to write text after `--init` and that text could do something, but that's a separate issue. Since `tsc --init` should do *something*, this issue is only about what that something is.

General consensus from the design meeting + external discussion:
 * No one likes the huge wall of text
 * Not many people like the commented-out options, and certainly not the "terrible idea" commented-out options
 * `"module": "commonjs"` is a hard no

Other live issues:
 * #39354 - set `types` to `[]` - yes
 * #49206, #39308 - increase default target - also yes
 * #39311 - set `udfcf` - I would argue moot by now since we should set the target high enough that this doesn't matter
 * #51207 - consult `package.json` - it's not obvious that your `package.json` is configured yet at the time you run `tsc --init`, this seems marginal

Other things we didn't get to:
 * `rootDir`, `outDir`: These are generally a good idea; no one likes the default side-by-side JS emit buuut there aren't strictly universal conventions here

Proposed new output:
```jsonc
{
    // For more info, see https://aka.ms/tsconfig
    // Or use the tsc-init wizard: https://aka.ms/tsc-init
    "compilerOptions": {
        "target": "es2022",
        "module": "nodenext",
        "rootDir": "./src",
        "outDir": "./dist",

        // For nodejs, add "node" to this array,
        // and remove "dom" from "lib"
        "types": [],
        "lib": ["dom", "es2022"],

        // Other outputs
        "sourceMap": true,
        // "declaration: true,

        // Typechecking options
        "noUncheckedIndexedAccess": true,
        "exactOptionalPropertyTypes": true,

        // Style options
        // "noImplicitReturns": true,
        // "noImplicitOverride": true,
        // "noUnusedLocals": true,
        // "noUnusedParameters": true,
        // "noFallthroughCasesInSwitch": true,
        // "noPropertyAccessFromIndexSignature": true,

        // We recommend all of these options
        "verbatimModuleSyntax": true,
        "isolatedModules": true,
        "moduleDetection": "force",
        "skipLibCheck": true,
        "strict": true
    }
}
```
I tried to order this from "most likely to edit" to "least likely to edit"

**Comment by andrewbranch** [comment]
Module options look good to me for Node.js / library authoring. I assume you’ve intentionally left in some redundancies so that good option _A_ stays set if someone changes/removes option _B_ that’s currently making _A_ redundant. For good measure, a list of those:

- `--module nodenext` implies `--target esnext`; you have it set to `es2022` (seems good to keep it set as is)
- `--module nodenext` implies `--moduleDetection force`
- `--verbatimModuleSyntax` implies `--isolatedModules` (people are more likely to turn off the former)

The one that’s missing from this good-but-redundant category is `--module nodenext` implies `--esModuleInterop`, which is a good idea to keep on, but the thing people are likely to switch to is `--moduleResolution bundler`, which implies `--allowSyntheticDefaultImports`, which is close enough. So not critical to add, I think.

TLDR looks good 👍

**Comment by DanielRosenwasser** [comment]
Though originally I was thinking more about `include` over `rootDir`, it makes sense given the guardrails `rootDir` provides if you screw something up? And that's important given people try to do custom `exclude`/`include` rules that are often unnecessary.

So I'm generally in favor around `outDir` and `rootDir`. The thing I am curious about is if we can detect sensible conventions such as folder name and existence of a `package.json`. This includes stuff like `src` as a well-known input directory, and special casing depending on if you're running from *inside* `src`.

**Comment by RyanCavanaugh** [comment]
👍 / 👎 ?
```
        // Other outputs
        // "declaration": true,
        // "sourceMap": true,
```

**Comment by andrewbranch** [comment]
Pretty much everybody is going to debug at some point, and it’s going to be bad until they realize they need to enable `sourceMap`

**Comment by fatcerberus** [comment]
The only thing that sucks about having options commented out is you can't hover on them to see the description of what they do, which doesn't help when you don't know what something does and want to know if you should enable it.  Unfortunately most options don't have an explicit setting that means "use the default behavior, whatever it happens to be today", so no easy solutions to be found here.

**Comment by Conaclos** [comment]
I could also suggest turning off `allowUnreachableCode`. It greatly help in some circumstance.

**Comment by System233** [comment]
My suggestion is that the tsconfig should only configure basic type checking and avoid trying to add things that users may need. Developers will refer to the documentation to configure what they really need; there's no need to overcomplicate things.

In China, there’s an idiom called “画蛇添足,” which literally means adding feet to a snake while drawing it. This idiom refers to the act of making unnecessary additions that complicate things rather than enhancing them. Adding a lot of options and comments that no one knows who will use or when they will be used exemplifies this unnecessary complication. Attempting to predict developers' environments to generate targeted default configurations and comments will never satisfy everyone. Since that’s the case, why not keep things a little simpler? Therefore, only the most basic configurations should be added, representing the minimal TS/JavaScript development environment. As for what specific options developers actually need, that’s not something we should be concerned about. Regarding the option comments in tsconfig, simply replacing them with a $schema field would suffice; don’t treat developers like idiots, managing them like a nagging mother.

**Comment by RyanCavanaugh** [comment]
```jsonc
{
    // For more info, see https://aka.ms/tsconfig
    "compilerOptions": {
        // File layout
        "rootDir": "./src",
        "outDir": "./dist",

        // Environment settings
        // See also https://aka.ms/tsconfig_modules
        "module": "nodenext",
        "target": "esnext",
        "types": [],
        // For nodejs:
        // "lib": ["esnext"],
        // "types": ["node"],
        // and npm install @types/node

        // Other outputs
        "sourceMap": true,
        // "declaration: true,

        // Stricter typechecking options
        // "noUncheckedIndexedAccess": true,
        // "exactOptionalPropertyTypes": true,

        // Style options
        // "noImplicitReturns": true,
        // "noImplicitOverride": true,
        // "noUnusedLocals": true,
        // "noUnusedParameters": true,
        // "noFallthroughCasesInSwitch": true,
        // "noPropertyAccessFromIndexSignature": true,

        // We recommend all of these options
        "strict": true,
        "verbatimModuleSyntax": true,
        "isolatedModules": true,
        "moduleDetection": "force",
        "skipLibCheck": true
    }
}
```

**Comment by DanielRosenwasser** [comment]
^ One issue with the above is that `rootDir` might still include files outside of `src`, but issue an error. Not sure if it's just worth specializing the error message there.

**Comment by justinfagnani** [comment]
I'd like to put in a pitch for `"outDir": "./"`

That's often the right value if you're migrating from JS and want to keep the same file layout, and it's also what you want to do if you have things like a top-level `index.js`, and non-distributed files like tests, demos, and local scripts.

Your src/ folder might look like:
```
src
├── demo
│   └── index.ts
├── lib
│   └── foo.ts
├── scripts
│   └── cleanup-things.ts
├── test
│   └── foo_test.ts
└── index.ts
```

And `index.js` and `lib/foo.js` would be the only output files "distributed".

**Comment by RyanCavanaugh** [comment]
Here's a little walkthrough of how we ended up here

Our general principles were:
 * Reduce the number of edits the median user has to make to get a "good" tsconfig
   * This might mean that no one has *zero* edits, but that very few people have to make a *lot* of edits
 * No more wall of text
 * Turn on options we think everyone should have on
 * Offer other options that are likely to be turned on
 * Use a more modern set of defaulted values

This link replaces having an enormous wall of text
```
{
  // For more info, see https://aka.ms/tsconfig
  "compilerOptions": {
```

While these are common, you do need to know what they are if you're using them, and `src` / `dist` are not universal naming conventions. `outDir` is also not needed if you're using `noEmit` or `bundler`
```
    // File layout
    // "rootDir": "./src",
    // "outDir": "./dist",
```

Most TS code is still frontend, so the default `dom` lib is a good choice. `types: []` is a big win for most projects and is an oft-missed override.

`esnext` target is a bit of a stretch. Ideally this would be something like "whatever `esnext` was two years ago", but in practice we don't expect people to be *unintentionally* using brand-new syntax without realizing it.
```
    // Environment settings
    // See also https://aka.ms/tsconfig_modules
    "module": "nodenext",
    "target": "esnext",
    "types": [],
    // For nodejs:
    // "lib": ["esnext"],
    // "types": ["node"],
    // and npm install -d @types/node
```

While you might not need declaration files at first, it's easier to keep your code in a declaration-friendly form than to try to add it later. Source maps are also generally useful
```
    // Other outputs
    "sourceMap": true,
    "declaration": true,
    "declarationMap": true,
```

Similarly, these options are annoying to enable later, but straightforward to comply with in a greenfield situation. It'e best to start with them on and turn them off if you don't like them
```
    // Stricter typechecking options
    "noUncheckedIndexedAccess": true,
    "exactOptionalPropertyTypes": true,
```

We expect most devs to turn on a few of these, but *which* few is very dependent on taste. Turning on all of them would be quite noisy by default (`noUnused*` generates a lot of erroneous squiggles mid-coding).
```
    // Style options
    // "noImplicitReturns": true,
    // "noImplicitOverride": true,
    // "noUnusedLocals": true,
    // "noUnusedParameters": true,
    // "noFallthroughCasesInSwitch": true,
    // "noPropertyAccessFromIndexSignature": true,
```

`strict: true` is obvious in modern TS
```
    // We recommend all of these options
    "strict": true,
```

`react-jsx` is a "best of all worlds" setting for the vast majority of code using JSX of any kind
```
    "jsx": "react-jsx",
```

Both these flags are modern best practice to enable third-party transpilers to take over for TS at any point. We strongly recommend both of them
```
    "verbatimModuleSyntax": true,
    "isolatedModules": true,
```

Inferring non-module scope for a file is a frequent source of confusion. Even though this flag is implied by `verbatimModuleSyntax`, it's included here to avoid having it accidently disabled.
```
    "moduleDetection": "force",
```

Errors from skipLibCheck are typically unactionable, instead pointing to misconfigurations in upstream packages. While it's better if your code *can* check with `skipLibCheck` off, in practice most projects will end up with this on for one reason or another (see #60427)
```
    "skipLibCheck": true
```
  }
}

Considered but not included:
 * `composite` is useful, but implies `incremental`, which in turn requires updating `.gitignore` to avoid checking in `tsbuildinfo`
 * `esModuleInterop` is already implied by any recommended module config, so is not needed
 * `erasableSyntaxOnly` bans `enum` so isn't likely to be popular for projects that don't need this particular constraint
 * The [set of flags to enable nodejs type stripping](https://nodejs.org/api/typescript.html#type-stripping) is fairly dangerous to offer to people who don't understand the caveats of that mode

**Comment by colinhacks** [comment]
While I agree `types: []` is a saner default, I'm don't think it's a pragmatic one. I'm perhaps in a unique position to evaluate the magnitude of disruption this would cause. I put together `@types/bun` when I was at Bun. Before that we published `bun-types` and recommended using `types` to load it. This was the case for the first 3 months post-Bun 1.0 and it was extremely problematic for our users. 

You can use our struggles here as a microcosm for the impact this change would have on the broader ecosystem. A sampler of the filed issues relating to this:

- https://github.com/oven-sh/bun/issues/463
- https://github.com/oven-sh/bun/issues/5674
- https://github.com/oven-sh/bun/issues/4890
- https://github.com/oven-sh/bun/issues/5983
- https://github.com/oven-sh/bun/issues/5134
- https://github.com/oven-sh/bun/issues/3030
- https://github.com/vercel/next.js/issues/58360

It's the kind of change that breaks ten thousand tutorials in a way where less experienced programmers won't know what went wrong. I'd expect this to result in a long-lived stream of "Cannot find name X" issues being opened in Node, Jest, React/ReactDOM, and other libraries that depend on global declaration loading.

Various highly upvoted issues caused by `"types"` inclusion in tsconfig.json:
- https://github.com/kulshekhar/ts-jest/issues/1068#issuecomment-549052448
- https://stackoverflow.com/a/54623931/1550155
- https://stackoverflow.com/a/69362236/1550155

I wasn't in the planning meetings so I don't know if the magnitude of this decision was fully considered. Based on my experience the disruption would be significant.

**Comment by RyanCavanaugh** [comment]
I hear what you're saying and I get the scenario, but conversely, we've dealt with *endless* problems dealing from automatic types inclusion, and those problems are much more difficult to address -- we can (and have) hardcoded things in error messages like "You need X in `types` to get Y`, but the problems that auto-inclusion presents are very tricky.

Non-strict package managers will lay out transitive dependencies in `@types/`, and since there's unfortunately no notion of a "transitive dev dependency", `@types` basically gets filled up with every type package that anyone ever thought about using. This leads to an extremely polluted files list, which is a big performance problem -- one team we investigated recently at Microsoft had *1,200* excess `types` being loaded. We don't have any way to solve this as it's not a per se misconfiguration - it's just bad.

Even huge companies don't know to do this by default; https://github.com/microsoft/TypeScript/issues/59780#issuecomment-2322862328, they cut their build time by over half by not auto-enumerating types

> Thanks @RyanCavanaugh, we did the changes in `tsconfig.base.json` to include only few types by default `"types": ["node", "jest", "@testing-library/jest-dom"]`, post this change and fixing missing the types the typecheck time has improved considerably from 70 mins to now nearly 33 mins.

This also makes it much more likely for lib pollution to occur -- people will be trying to target ES2018, but one transitive transitive transitive type dependency will pull in `/// <reference lib="es2022">` for some reason. Again, there's no solution here, for all intents and purposes this isn't an error but it still causes TS to miss accidental use of the higher-target APIs.

Maybe this is just a "grass is greener on the other side" but I'd like to try this and see if it's truly worse or not. We've been dealing with `types` auto-enumeration for a very long time and have accumulated a lot of issues that are solved by setting it to `[]`; it's at the very least worth the experiment. We can always change `tsc --init` again.

Telling people to fix their configs with `types: []` is one of the most common config-related things we do here. I ran out of steam looking through search results so here are just a few of the hundreds of times this comes up

* #5509
* #9731
* #11257
* https://github.com/microsoft/TypeScript/issues/11781#issuecomment-503773748
* #11948
* #18588
* #19910
* https://github.com/microsoft/TypeScript/issues/34576#issuecomment-544438491
* #40388
* #49283
* #59873

**Comment by colinhacks** [comment]
Thanks @RyanCavanaugh! That certainly helps me understand the color of the grass on your side. Just wanted to make sure the other side of this debate got its due. I agree that its the better call and even if this causes some initial disruption, sounds like it's the better move on a long enough time horizon.

**Comment by aweebit** [comment]
Here is my suggestion:

```jsonc
{
  // For more info, see https://aka.ms/tsconfig
  "compilerOptions": {
    // File layout
    // "rootDir": "./src",
    // "outDir": "./dist",

    // Environment settings
    // See also https://aka.ms/tsconfig_modules
    "module": "nodenext",
    "target": "esnext",
    "types": [],
    // For nodejs:
    // "lib": ["esnext"],
    // "types": ["node"],
    // and npm install -D @types/node

    // Other outputs
    "sourceMap": true,
    "declaration": true,
    "declarationMap": true,

    // Incremental builds
    "incremental": true,
    "tsBuildInfoFile": "./node_modules/.tsbuildinfo/tsconfig.tsbuildinfo",

    // Stricter typechecking options
    "noUncheckedIndexedAccess": true,
    "exactOptionalPropertyTypes": true,

    // Style options
    // "noImplicitReturns": true,
    // "noImplicitOverride": true,
    // "noUnusedLocals": true,
    // "noUnusedParameters": true,
    // "noFallthroughCasesInSwitch": true,
    // "noPropertyAccessFromIndexSignature": true,

    // We recommend all of these options
    "checkJs": true,
    "strict": true,
    "jsx": "react-jsx",
    "useDefineForClassFields": true,
    "verbatimModuleSyntax": true,
    "isolatedModules": true,
    "moduleDetection": "force",
    "noUncheckedSideEffectImports": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true
  }
}
```

#### The changes (from least to most arguable):

- `npm install -d` is for debugging 😛 `-D` has to be uppercase
- Enabled `forceConsistentCasingInFileNames`. Why on earth would anyone want inconsistent casing?
- Enabled `noUncheckedSideEffectImports`. Obviously the right thing to do in all new projects
- Enabled `useDefineForClassFields`. It is already implied by `"target": "esnext"`, but it's reasonable to keep it on even if `target` is set to a different value after initialization
- Enabled `checkJs`. Any reasons not to? It's great for checking config files that have to be written in JavaScript. Also, if you use `noEmit` and don't have `outDir` set up because you transpile with some other tool, it can help you realize that you forgot to specify that tool's output directory in `"exclude"` because you will see a bunch of errors from the output .js files.
- Enabled `incremental` with `tsBuildInfoFile` set to a hidden path within `node_modules`. Since `.gitignore` should already include `node_modules`, there is no need to additionally adjust it. Vite does a similar thing by using `node_modules/.tmp` as the output directory for `.tsbuildinfo` files.

I would also support enabling `erasableSyntaxOnly`. I think use of features that are not type-level extensions of JavaScript should be discouraged in new projects.

#### Update

Adding an extra comment line here might be a good idea:

```jsonc
    // Environment settings
    // See also https://aka.ms/tsconfig_modules
    "module": "nodenext",
    // For browser: <----------
    "target": "esnext",
    "types": [],
    // For nodejs:
    // "lib": ["esnext"],
    // "types": ["node"],
    // and npm install -D @types/node
```

This way, the options that have to be replaced for Node are clearly separated from `"module": "nodenext"` that should stay, so it's less likely it will be deleted by accident or misunderstanding.

**Comment by Renegade334** [comment]
> This also makes it much more likely for lib pollution to occur -- people will be trying to target ES2018, but one transitive transitive transitive type dependency will pull in `/// <reference lib="es2022">` for some reason. Again, there's no solution here, for all intents and purposes this isn't an error but it still causes TS to miss accidental use of the higher-target APIs.

It's somewhat unfortunate that there's no way for a library to indicate a transitive lib dependency for its exported definitions, other than `/// <reference lib="..." />` to surreptitiously yeet it into the entire compilation, or \[ silently expose dependents to unexpected `any` in arbitrary places \| generate compilation errors from node_modules \] depending on the value of skipLibCheck. Equally, there's not currently any way for downstream consumers to opt in or out.

Maybe there's scope in the future for:
- some kind of compiler option which turns lib directives into lib _assertions_, such that the directive itself doesn't cause the lib to be imported, but if any source file contains a directive pointing to a lib that isn't loaded, the compiler reports an error
- alternatively, a new directive that provides the same functionality on an individual basis (`/// <reference depends="..." />` or similar)

**Comment by RyanCavanaugh** [comment]
> npm install -d is for debugging 😛 -D has to be uppercase

Thank you 😁

> Enabled forceConsistentCasingInFileNames

This is on by default now

> Enabled noUncheckedSideEffectImports

Yep, this was also brought up offline. Added to PR.

> Enabled checkJs

This is a bit more invasive and has implications in module resolution that aren't ideal. This makes it more likely to end up accidently consuming an output vs an input and I don't think is a great idea.

The others were also discussed offline and were close calls, but we generally wanted to keep this output shorter so erred on the side of leaving them out. `erasableSyntaxOnly` is a really tough call because `enum` is so well-loved but not supported - if this moves through committee we'd probably turn it on.

**Comment by lishaduck** [comment]
> The others were also discussed offline and were close calls, but we generally wanted to keep this output shorter so erred on the side of leaving them out.

Wanted to add that I'd also be very happy to see `incremental` + `tsbuildinfofile` added. If it's too long, could changing the `tsbuildinfofile` default in v6 be an option?

> `erasableSyntaxOnly` is a really tough call because `enum` is so well-loved but not supported - if this moves through committee we'd probably turn it on.

Would https://github.com/microsoft/TypeScript/pull/61414 change that stance?
I do miss `enum`s, but `erasableSyntaxOnly` is great.

(I primarily care about the default settings because I need to convince people that e.g., `"module": "nodenext"` is TS team recommended or else they just stick with `"module": "node"`)

**Comment by lishaduck** [comment]
> We expect most devs to turn on a few of these, but which few is very dependent on taste. Turning on all of them would be quite noisy by default (noUnused* generates a lot of erroneous squiggles mid-coding).

While I personally find it very noisy (as a default, I do use it), it's notable that `deno-ts` turns on `noImplicitOverride` by default now. Might be worth aligning or at least checking their rational.
Tracking issue: https://github.com/denoland/deno/issues/25162. `noImplicitOverride` issue: https://github.com/denoland/deno/issues/11836

**Comment by RyanCavanaugh** [comment]
Putting the flag in the compiler in the first place (and not deprecating it at the first available opportunity) is our endorsement of something as a reasonable idea.

Particularly for the style ones, we're super uninterested in having an unendable debate about which of the six listed are A Good Idea For Everyone and which are Not Idiomatic TypeScript Actually. If they're all off, they're all off. If two of the six are on, then everyone is here arguing that it's the wrong two.

**Comment by ljharb** [comment]
Is there a flag I can pass along with `tsc --init` that includes the full list of options and explanations in the generated config file?

**Comment by endjynn** [comment]
Please consider adding an option to generate a FULL tsconfig.json, the same as the previous behavior. I found this feature incredibly useful to keep track of all the various tsconfig options. We're developers and reading code is what we do. I want to be able to easily read all the options as code (well, as JSON) rather than having to refer to the documentation.

**Comment by RyanCavanaugh** [comment]
I'm not really sure why `tsc --init` is the only place in the whole project where we should provide an exhaustive in-situ summary of stuff that's already in the docs. It seems way more tractable to improve  https://www.typescriptlang.org/tsconfig/

**Comment by ljharb** [comment]
@RyanCavanaugh It's very useful to have an exhaustive, editable, version-controlled file where I can tweak settings. Having a minimal set requires me to do my own research before I even can know that I'm missing information.

**Comment by RyanCavanaugh** [comment]
What specifically makes exhaustiveness useful? I'm thinking of rarely-set, never-recommended things like `suppressImplicitAnyIndexErrors`, `charset`, etc - what's the value in having those in your config?

Conversely, what do you do when new options appear?

**Comment by endjynn** [comment]
> [@RyanCavanaugh](https://github.com/RyanCavanaugh) It's very useful to have an exhaustive, editable, version-controlled file where I can tweak settings. Having a minimal set requires me to do my own research before I even can know that I'm missing information.

Yes this. Having easy access to a *full* and documented tsconfig.json allows me to init a config when TypeScript changes versions. I can then immediately see which new options have been added instead of having to trawl through release pages or documentation.
It's massively useful to be able to generate a full/complete tsconfig.json using just tsc.

**Comment by RyanCavanaugh** [comment]
Interesting. So your workflow is this?

1. `tsc --init`
2. Change whatever you need changed
3. On version update, delete `tsconfig.json`, re-run `tsc --init`
4. Re-apply the changes from step 2
5. git diff, see what the new compiler options are, or if their descriptions have changed?
6. git add, git commit

**Comment by ljharb** [comment]
@RyanCavanaugh yes, basically. it'd be easier if there was a "flesh out" command that just added any missing options with their defaults, but otherwise that workflow is the only option.

**Comment by endjynn** [comment]
> Interesting. So your workflow is this?
> 
>     1. `tsc --init`
> ...

Yes, or something similar :)
It allows us to diff tsconfig.json between TypeScript versions to easily see what has changed.

**Comment by RyanCavanaugh** [comment]
It's less work and more instructive to run
```
tsc --help --all > ts-config.md
```
and diff that artifact instead

**Comment by endjynn** [comment]
> It's less work and more instructive to run
> 
> ```
> tsc --help --all > ts-config.md
> ```
> 
> and diff that artifact instead

This is an OK work-around. It is useful to be able to compare tsconfig.json configurations directly since those are in direct use by the project. I could keep a version controlled ts-config.md pinned to my project TypeScript version but it's not as elegant, as simple or as immediately "grokable" as a full tsconfig.json output.

Still, it's a possible work-around if we can't get the old behavior back. Thank you :)

**Comment by ljharb** [comment]
Is there any similar output that's programmatically readable? JSON perhaps?

**Comment by ljharb** [comment]
Also, the help output does not include the default value of each configuration option, so it's unfortunately useless to me :-/

**Comment by RyanCavanaugh** [comment]
> Also, the help output does not include the default value

<img width="1092" height="410" alt="Image" src="https://github.com/user-attachments/assets/ef6e348b-cab7-48e0-9b24-b03482452cba" />

?

**Comment by ljharb** [comment]
huh! my bad, the markdown parser i chose must not be working properly. Thanks!

**Comment by StoneCypher** [comment]
@RyanCavanaugh - sorry, i know this is weeks late, but could we have an --init-full or something that had the old behavior?

**Comment by RyanCavanaugh** [comment]
`npx -p typescript@5.7 tsc --init` (jk)

Feel free to log a new suggestion for the "full" mode
